sqlmesh/core/macros.py

```diff
@@ -414,9 +418,16 @@ def columns_to_types(self, model_name: TableName | exp.Column) -> t.Dict[str, ex
             default_catalog=self.default_catalog,
             dialect=self.dialect,
         )
-        columns_to_types = self._schema.find(
-            exp.to_table(normalized_model_name), ensure_data_types=True
+        model_name = exp.to_table(normalized_model_name)
+
+        columns_to_types = (
+            self._schema.find(model_name, ensure_data_types=True) if self._schema else None
         )
+        if columns_to_types is None:
```

Comment:
I'd also add an explicit unit test for this and not just rely on the sushi project to test this.

Reply:
Good idea, will add shortly. Thanks for reviewing 👍

Reply:
Added - https://github.com/TobikoData/sqlmesh/pull/3006/commits/c21c23b3aae6860221ecafeadf4d1e5ce285da89